app/Http/Controllers/DuesTransactionController.php

```diff
@@ -116,12 +116,23 @@ public function store(Request $request)
             }
         }
 
-        //Check to make sure there isn't already an existing package for the target user
-        $existingTransaction = DuesTransaction::where('dues_package_id', $request->input('dues_package_id'))
-            ->where('user_id', $request->input('user_id'))->first();
-        if ($existingTransaction) {
-            return response()->json(['status' => 'error',
-                'message' => 'There is already a pending Dues Transaction for this user', ], 400);
+        // If there's an existing active transaction that hasn't been paid, delete it
+        // and replace it with the one currently being requested
+        if ($user->dues->count() > 0) {
+            $existingTransaction = $user->dues->last();
+            $pkgIsActive = $existingTransaction->package->is_active;
+            if ($pkgIsActive) {
+                $hasPayment = $existingTransaction->payment()->exists();
+                if ($hasPayment) {
+                    $paymentAmount = $existingTransaction->payment->sum('amount');
+                    $paidTotal = ($paymentAmount >= $existingTransaction->getPayableAmount());
+                    if (! $paidTotal) {
```

Comment:
If they paid anything, IMO we shouldn't be deleting it because that's money. I get that it's only soft delete, but still.

Reply:
I agree. In what case will the payment amount be less than the total?

Reply:
Can we leave partially paid transactions?

Reply:
It is leaving partially paid transactions now.

Reply:
Ok, this seems fine.